crates/pg_workspace_new/src/workspace/server.rs

```diff
@@ -342,6 +342,45 @@ impl Workspace for WorkspaceServer {
             skipped_diagnostics: 0,
         })
     }
+
+    #[tracing::instrument(level = "info", skip(self))]
+    fn get_completions(
+        &self,
+        params: super::CompletionParams,
+    ) -> Result<pg_completions::CompletionResult, WorkspaceError> {
+        let doc = self
+            .documents
+            .get(&params.path)
+            .ok_or(WorkspaceError::not_found())?;
+
+        tracing::info!("Found the document.");
+        tracing::info!("Looking for statement at position: {:?}", &params.position);
+
+        let statement_ref = match doc.statement_ref_at_offset(&params.position) {
+            Some(s) => s,
+            None => return Ok(pg_completions::CompletionResult::default()),
+        };
+
+        let tree = self.tree_sitter.fetch(&statement_ref);
+        let text = doc
+            .statement_by_id(statement_ref.id)
+            .expect("Found statement_ref but no matching statement")
+            .text;
+
+        let schema_cache = self
+            .schema_cache
+            .read()
+            .map_err(|_| WorkspaceError::runtime("Unable to load SchemaCache"))?;
```

Comment:
I agree that the document api is a bit weird, but you should be able to query for `statement_at_offset`, and the `Statement` also includes the `StatementRef`. Open for ideas to improve that api.

Reply:
you're right! thanks. 

Hmm I think the only problem is the name, `StatementRef` sounds to me like a `&Statement` of some kind. 

Maybe the `StatementRef` could be the actual `Statement`, and the doc could have a `get_text(s: &Statement)` method, since the text is probably the only thing that callers care about

Reply:
Good point! I like that proposal. Will try it out